app/src/main/java/com/example/itunesapp/viewModel/ItunesViewModel.kt

```diff
@@ -1,46 +1,19 @@
 package com.example.itunesapp.viewModel
 
-import android.app.Application
-import android.content.Context
-import android.util.Log
-import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
-import com.example.itunesapp.database.DatabaseImpl
-import com.example.itunesapp.model.ResultDatabase
-import com.example.itunesapp.network.ApiInstance
-import com.example.itunesapp.repositoryMapping.ResultMappingImpl
-import kotlinx.coroutines.GlobalScope
+import androidx.lifecycle.viewModelScope
+import com.example.itunesapp.model.RemoteDataModel
+import com.example.itunesapp.repository.Repository
 import kotlinx.coroutines.launch
 
-class ItunesViewModel(application: Application) : AndroidViewModel(application) {
+class ItunesViewModel (private val repository: Repository): ViewModel() {
 
+    var searchSongsList:MutableLiveData<ArrayList<RemoteDataModel>> = MutableLiveData()
 
-    val searchItunesApi = ApiInstance.searchItunesApi
-
-    var databaseImpl:DatabaseImpl = DatabaseImpl.getDatabase(application)
-
-    val resultMappingImpl = ResultMappingImpl()
-
-
-    fun searchItem(){
-
-        GlobalScope.launch {
-            val result = searchItunesApi.searchSongs("#search")
-            if (result != null){
-                var searchItem = result.body()
-                if (searchItem != null) {
-
-                    var searchList:List<ResultDatabase> = resultMappingImpl.searchList(searchItem.results)
-                    Log.d("View Model@@@@@@@@@@",searchList.toString())
-                    databaseImpl.databaseDao().addSearchItems(searchList)
-                }
-
-
-            }
+    fun search(name: CharSequence?) {
+        viewModelScope.launch {
+            searchSongsList.value = repository.search(name as String)
```

Comment:
Never cast nullable type to a non nullable one. This goes again the null safety kotlin tries to provide.

If you need a non null String, only accept a non null string in the function.